dropwizard-timer/src/main/java/smartthings/dw/timer/MethodInvocationTimingInterceptor.java

```diff
@@ -0,0 +1,42 @@
+package smartthings.dw.timer;
+
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.annotation.Timed;
+import com.google.inject.Inject;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+
+import java.lang.reflect.Method;
+
+public class MethodInvocationTimingInterceptor implements MethodInterceptor {
+
+    private MetricRegistry registry;
+
+    @Inject
+    public void setRegistry(MetricRegistry registry) {
+        this.registry = registry;
+    }
+
+    @Override
+    public Object invoke(MethodInvocation invocation) throws Throwable {
+        Timed timedAnnotation = invocation.getMethod().getAnnotation(Timed.class);
+        String name = findName(timedAnnotation.name(), invocation.getMethod());
+
+        Timer.Context ctx = registry.timer(name).time();
+
+        try {
+            return invocation.proceed();
+        } finally {
+            ctx.stop();
+        }
+    }
+
+    private String findName(String name, Method method) {
+        if (name == null || name.isEmpty()) {
+            return method.getName();
```

Comment:
Not sure if this is available off this AOP MethodInvocation instance, but it might be cool to append the Class name as well.